### PR TITLE
Feature/reward points customer

### PR DESCRIPTION
**Pull request by joshua1234511**

### CODE REVIEW

File: application/config/config.php [hunk 1]
```diff
@@ -358,7 +358,7 @@
 | https://codeigniter.com/user_guide/libraries/encryption.html
 |
 */
-$config['encryption_key'] = '';
+$config['encryption_key'] = 'ec58a27d5854523044e3283b673a97c1deb59fe1e6f2ca7f77ed5e0f4c5aa02a';
```
Comment (daN4cat) [inline application/config/config.php]: please remove your key

**Comment by jekkos** [comment]
Cool to see those PR's coming in. On the other hand I think we should consider doing a small release first as we have quite some functionality already?

**Comment by daN4cat** [comment]
We can pull this in and wait for the tax rework by @SteveIreland. Also I need to test and polish a bit things before we can baseline and release.

**Comment by jekkos** [comment]
With polish you mean refactor? Maybe we might want to make an epic for that

**Comment by daN4cat** [comment]
I mean adjust few things or inconsistencies. You can call it small refactoring too.

File: application/config/config.php (hunk 1 above)
Comment (joshua1234511) [inline application/config/config.php]: Done Removed key and pushed

**Comment by joshua1234511** [comment]
I agree on refactor as the language files as i see are in proper order  i did not add properly .

I am working now on adding a front end for customers where they can add id or email  with date birth and be able to check their sale and Reward points available.

**Comment by daN4cat** [comment]
And how do you plan to interface OSPOS from the mobile app?

**Comment by joshua1234511** [comment]
@daN4cat 
Did not understand interface OSPOS from the mobile app?
There is a mobile app also

**Comment by daN4cat** [comment]
How your mobile app will call OSPOS service?

**Comment by jekkos** [comment]
@joshua1234511 would be interested to see whether you're gonna open source it or not?

**Comment by joshua1234511** [comment]
Yes it would be open source

**Comment by joshua1234511** [comment]
Oh by  mobile I did not mean the app 
It was the login using e-mail and mobile number for validation of customer

**Comment by daN4cat** [comment]
Do you envision a solution where the customer could login to a different OSPOS page? Is that what you mean?

If yes I would be very careful as you open a backdoor for potential attacks. There are security implications to consider and talk about.

**Comment by joshua1234511** [comment]
Yes something like that 
Yes have added a new controller customer_login and managing through that. 
For customer added a new user-data customer_id for handling the security side
Pushed the new changes to new commit 

Attached a new Screen cast with new features
: https://www.screencast.com/t/lZbtEJidV

Admin:         /login
Customer:   /customer_login

**Comment by joshua1234511** [comment]
Please review the code and suggest if any security issues

**Comment by daN4cat** [comment]
Thank for the extra contribution but please revert to the previous version without the customer login.
I prefer to keep that separate and not have everything in one big pot.

**Comment by joshua1234511** [comment]
@daN4cat I have reverted back changes to the last commit and pushed
ie: the previous version without the customer login Changes

**Comment by daN4cat** [comment]
@joshua1234511 thanks for reverting the commit.

Please add all the *.sql content to the upgrade script 3.0.2_to_3.1.0.sql, also you need to make sure tables.sql contains the new tables as a new installation needs to be fully correct from an SQL schema point of view.

**Comment by joshua1234511** [comment]
Added the sql  to upgrade script 3.0.2_to_3.1.0.sql
And Also modified the tables sql to incorporate these table changes

**Comment by jlctmaster** [comment]
Hello @joshua1234511 , thanks for this improvement, @daN4cat  who should apply the translations? For example, I use the module translated into Spanish, can I do it?

**Comment by daN4cat** [comment]
@jlctmaster you can using weblate, check the wiki for instructions.

**Pull request merged**